### PR TITLE
Add macOS example app

### PR DESCRIPTION
**Pull request by jfedak**
## Summary
This PR adds a macOS app using React Native (see [react-native-macos docs](https://microsoft.github.io/react-native-windows/docs/rnm-getting-started))

## Test plan
This app builds and runs on macOS. In order to run this app on your own, follow `README.md`.

### CODE REVIEW

File: MacOSExample/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+open macos/MacOSExample.xcworkspace
+```
+
+or build and run directly from the command line:
+
+```
+yarn react-native run-macos
+```
+
+### Troubleshoot
```
Comment (tomekzaw) [inline MacOSExample/README.md:58]:
```suggestion
### Troubleshooting
```

File: MacOSExample/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+If you have trouble with building the app, try running
```
Comment (tomekzaw) [inline MacOSExample/README.md:60]:
```suggestion
If you have trouble with building the app, try running

```

File: MacOSExample/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+```
+yarn react-native run-macos
+```
+
+### Troubleshoot
+
+If you have trouble with building the app, try running 
+```
+rm -rf Pods build Podfile.lock
```
Comment (tomekzaw) [inline MacOSExample/README.md:62]:
```suggestion
rm -rf Pods Podfile.lock build
```

**Comment by tomekzaw** [comment]
Works for me ✅ 

screenshot

**Review by tomekzaw — CHANGES_REQUESTED** [review]

**Review by tomekzaw — APPROVED** [review]

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**